Fixes multiple bugs

hbcarlos opened the pull request:
Necessary to fix tests in JupyterLab.

# Code changes:
* Dependency process: When linking `@jupyter/ydoc` to jupyterlab in dev mode, JupyterLab complains about ydoc missing this dependency. I added it in devDependencies so we only install it in dev mode.
* I changed the argument of the static method `YNotebook.create` to be able to initialize the notebook in a generic way. For example, JupyterLab needs to initialize the model with `nbformat`, `nbformat_minor`, and `metadata` but other frontends might need other attributes.
* Fixed the imports
* Fixed the callbacks for metadata changed in YCell and YNotebook. Both `oldValue` and `newValue` might be `undefined`.

fcollonval commented on javascript/src/api.ts:
```diff
@@ -388,9 +388,9 @@ export namespace ISharedNotebook {
+     * The content of the notebook.
      */
-    languagePreference?: string;
+    data?: Partial<nbformat.INotebookContent>;
```
Could you a test for this?

hbcarlos replied:
There is a new section `#factory` with these tests.

hbcarlos commented on javascript/package.json:
```diff
@@ -58,7 +58,8 @@
         "rimraf": "^3.0.0",
         "typedoc": "^0.23.21",
-        "typescript": "^4.8.0"
+        "typescript": "^4.8.0",
+        "process": "0.11.10"
```
Basically, `@jupyter/ydoc` is a package created to be consumed by JupyterLab. For development purposes, we need to `yarn link "@jupyter/ydoc"` in JupyterLab. When building JupyterLab with `@jupyter/ydoc` linked, JupyterLab complains about `@jupyter/ydoc` missing `process/browser` and the fix is to add process to the dev dependencies of `@jupyter/ydoc`.

fcollonval replied:
Thanks for the comment.

hbcarlos commented:
Regarding the tests, I moved them to different files. The only changes are that I added `YNotebook/factory` as you requested. In addition, I added a few tests for the YFile.

fcollonval reviewed (approved):
Thanks @hbcarlos
